source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRTableCellView.mm

```diff
@@ -0,0 +1,85 @@
+//
+//  ACRTableCell.m
+//  ACRTableCell
+//
+//  Copyright © 2021 Microsoft. All rights reserved.
+//
+
+#import "ACRTableCellView.h"
+#import "ACOBaseCardElementPrivate.h"
+#import "ACOHostConfigPrivate.h"
+
+@implementation ACRTableCellView {
+    __weak UIView *_contentView;
+    ACRTableCellDefinition *_definition;
+}
+
+- (instancetype)init:(ACOBaseCardElement *)baseCardElement
+      cellDefinition:(ACRTableCellDefinition *)definition
+            rootView:(ACRView *)rootView
+              inputs:(NSMutableArray *)inputs
+          hostConfig:(ACOHostConfig *)acoConfig;
+{
+    self = [super init];
+    if (self) {
+
+        [[ACRTableCellRenderer getInstance] render:self rootView:rootView inputs:inputs baseCardElement:baseCardElement hostConfig:acoConfig];
+        _definition = definition;
+        if ([self.subviews count]) {
+            _contentView = self.subviews[0];
+        }
+        [self addSubview:_contentView];
+        [self setAutoLayout];
+        [self configureForStyle:acoConfig];
+    }
+    return self;
+}
+
+- (void)setAutoLayout
+{
+    if (_contentView) {
+        NSLayoutConstraint *contentViewWidthAnchor0 = [_contentView.widthAnchor constraintLessThanOrEqualToAnchor:self.widthAnchor];
```

Comment:
It's debatable. I agree it's cleaner that way. Apple's API itself is verbose. It takes 5 parameters to set one constraint. you have to push 5 variables to call the new function, and there is variation in Apple's API for setting constraints. we would have to create a new function to account for that. What if we want to use constant + anchors instead of just anchors.

Reply:
I'm talking as a person who is reading the code, it took me longer than expected to figure out that there was no connection between all the constraints and that there were 4 instead of 2

Reply:
i will make the change in 2/3 pr